src/components/button/button.d.ts

```diff
@@ -14,7 +14,7 @@ export interface ButtonProps {
   children?: React.ReactNode;
   renderRouterLink?: (args: object) => React.ReactNode;
   forwardRef?: () => void;
-  onClick?: () => void;
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
```

Comment:
Button can sometimes be an `a` element instead of `button`.
https://github.com/Sage/carbon/blob/35cf0e369a982e63b5a271f22fcc5eb027b50852/src/components/button/button.component.js#L26